### PR TITLE
initial commit of the work to add the middleware

### PR DESCRIPTION
**Pull request by bspink**
gettin' things going

### CODE REVIEW

File: http_restrict/middleware/__init__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            raise MiddlewareNotUsed
+
+        self.allowed_methods = allowed_methods
+
+    def process_request(self, request):
+
+        if request.method in self.allowed_methods:
+            return None
+
+        return HttpResponse(status=501)
```
Comment (bspink) [inline http_restrict/middleware/__init__.py]: Hey moron, this should be 405 not 501!

**Comment by fud** [comment]
:oden:  I like that we're open sourcing this. :dango:

**Comment by bspink** [comment]
LGTM

**Comment by bspink** [comment]
LGTM

**Comment by prologic** [comment]
LGTM

**Pull request merged**